src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/ExternalLanguageOperations.cs

```diff
@@ -0,0 +1,343 @@
+        private const string LanguageNameParamName = "LANGUAGE";
+
+        private string GetDropScript(string languageName)
+        {
+            return $@"{DropScript} [{CUtils.EscapeStringCBracket(languageName)}]";
+        }
+
+        /// <summary>
+        /// Returns the status of external languages in a connection
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="languageName"></param>
+        /// <returns></returns>
+        public virtual bool GetLanguageStatus(IDbConnection connection, string languageName)
+        {
+            bool status = false;
+            try
+            {
+                using (IDbCommand command = connection.CreateCommand())
+                {
+                    command.CommandText = StatusScript;
+                    var parameter = command.CreateParameter();
+                    parameter.ParameterName = $"@{LanguageNameParamName}";
+                    parameter.Value = languageName;
+                    command.Parameters.Add(parameter);
+                    using (IDataReader reader = command.ExecuteReader())
+                    {
+                        while (reader.Read())
+                        {
+                            status = (Convert.ToBoolean(reader[0].ToString()));
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.Write(TraceEventType.Warning, $"Failed to get language status for language: {languageName}, error: {ex.Message}");
+                status = false;
+            }
+
+            return status;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <returns></returns>
+        public virtual List<ExternalLanguage> GetLanguages(IDbConnection connection)
+        {
+            return GetLanguages(connection, null);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="languageName"></param>
+        /// <returns></returns>
+        public virtual ExternalLanguage GetLanguage(IDbConnection connection, string languageName)
+        {
+            List<ExternalLanguage> result = GetLanguages(connection, languageName);
+            if (result != null && result.Any())
+            {
+                return result.First();
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="language"></param>
+        public virtual void UpdateLanguage(IDbConnection connection, ExternalLanguage language)
+        {
+            if (string.IsNullOrWhiteSpace(language?.Name))
+            {
+                throw new LanguageExtensibilityException($"Invalid language. name: {language?.Name}");
+            }
```

Comment:
nit: This is never going to print anything useful for language name since it's always empty or null